### ISSUE TITLE
`@JsonView` does not work on class-level for records

### ISSUE THREAD

**Issue by u3r** [FasterXML/jackson-databind#4085]
Jackson(-Annotations) 2.15.2

```java
@JsonView( Views.Default.class )
public record CollectionResponse<T>( Long totalItems,  Long currentPage, Long totalPages, Long pageSize, List<T> items) { }
```
Does always render to an empty json object `{}` 
whereas 
```java
public record CollectionResponse<T>(@JsonView( Views.Default.class ) Long totalItems, @JsonView( Views.Default.class ) Long currentPage,
                                    @JsonView( Views.Default.class ) Long totalPages, @JsonView( Views.Default.class ) Long pageSize,
                                    @JsonView( Views.Default.class ) List<T> items) {
```
Give the expected output (given e.g. a Spring Rest-Endpoint annotated with `@JsonView( Views.Default.class )` 

While the javadoc does not mention records it mentiones pojos:

> Starting with 2.9, it is also possible to use this annotation on POJO classes to indicate the default view(s) for properties of the type, > unless overridden by per-property annotation.

Not sure if this is a feature request or a bug - but either way I feel this should include records.

**Comment by cowtowncoder** [comment]
It does sound like a bug if and when matching POJO would correctly associate class-level annotation as the default for properites.

**Comment by cowtowncoder** [comment]
Adding passing unit test via #4551 -- as far as I can see this works for 2.18 (and if I ran test correctly, 2.17 branch as well).
So would need Jackson-only reproduction against 2.17.1.

**Comment by cowtowncoder** [comment]
Marking as fixed in 2.18.0, closing; may be re-opened with a failing test that does not rely on other frameworks or libraries.
